XmlNode.h

```diff
@@ -60,6 +60,40 @@ struct XmlError
   string text;
 };
 
+// ****************************************************************************
+// Classes for custom XML rlated exceptions.
+// ****************************************************************************
+
+class XmlException : public std::exception
```

Comment:
Move the XML exceptions to a separate file `XmlExceptions.h`. The general guideline is to have one class per translation unit, although in this case I think it makes sense to have all the `XmlException`-derived classes in the same file.

I would generally recommend moving the implementation to a separate CPP file. If you choose not to (since it is very brief), make sure you mark the methods as `inline`, otherwise we may get compilation errors if the header is included in multiple files.

Reply:
Done